### PR TITLE
Add support for generic type - function type

### PR DESCRIPTION
**Pull request by agankarin**
Similar to function, add ability to declare function types with generic syntax

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -177,6 +177,7 @@ export interface FunctionType {
     kind: "function-type";
     parameters: Parameter[];
     returnType: Type;
+    typeParameters: TypeParameter[];
```
Comment (RyanCavanaugh) [inline lib/index.ts]: Please make this optional so it's not a breaking change

Comment (RyanCavanaugh) [reply lib/index.ts]: Never mind

**Comment by RyanCavanaugh** [comment]
Published `3.4.0`. Thanks!

**Comment by agankarin** [comment]
Thank you very much!

**Pull request merged**